Kernel/Devices/Audio/Management.cpp

```diff
@@ -38,23 +39,33 @@ UNMAP_AFTER_INIT AudioManagement::AudioManagement()
 
 UNMAP_AFTER_INIT void AudioManagement::enumerate_hardware_controllers()
 {
-    if (!PCI::Access::is_disabled()) {
-        MUST(PCI::enumerate([&](PCI::DeviceIdentifier const& device_identifier) {
-            // Note: Only consider PCI audio controllers
-            if (device_identifier.class_code().value() != to_underlying(PCI::ClassID::Multimedia)
-                || device_identifier.subclass_code().value() != to_underlying(PCI::Multimedia::SubclassID::AudioController))
-                return;
+    if (PCI::Access::is_disabled())
+        return;
+    MUST(PCI::enumerate([&](PCI::DeviceIdentifier const& device_identifier) {
+        // Only consider PCI multimedia devices
+        if (device_identifier.class_code().value() != to_underlying(PCI::ClassID::Multimedia))
+            return;
 
-            dbgln("AC97: found audio controller at {}", device_identifier.address());
-            auto ac97_device = AC97::try_create(device_identifier);
-            if (ac97_device.is_error()) {
-                // FIXME: Propagate errors properly
-                dbgln("AudioManagement: failed to initialize AC97 device: {}", ac97_device.error());
-                return;
+        auto create_audio_controller = [](PCI::DeviceIdentifier const& device_identifier) -> ErrorOr<NonnullLockRefPtr<AudioController>> {
+            switch (static_cast<PCI::Multimedia::SubclassID>(device_identifier.subclass_code().value())) {
+            case PCI::Multimedia::SubclassID::AudioController:
+                return AC97::try_create(device_identifier);
+            case PCI::Multimedia::SubclassID::HDACompatibleController:
+                return Audio::IntelHDA::Controller::create(device_identifier);
+            default:
+                return ENOTSUP;
             }
-            m_controllers_list.append(ac97_device.release_value());
-        }));
-    }
+        };
+
+        dbgln("AudioManagement: found audio controller {} at {}", device_identifier.hardware_id(), device_identifier.address());
+        auto audio_controller_device = create_audio_controller(device_identifier);
+        if (audio_controller_device.is_error()) {
+            // FIXME: Propagate errors properly
```

Comment:
I don't think there's really anything to propagate - we catch this error here and that's actually OK.

Reply:
Agreed, we might want to this in the future when we want to keep track of/deal with device issues but the way it is implemented right now, doesn't require a `FIXME`. Thanks!